src/snowflake/connector/connection_diagnostic.py

```diff
@@ -0,0 +1,578 @@
+import base64
+import ipaddress
+import json
+import os
+import re
+import socket
+import ssl
+import tempfile
+import OpenSSL
+
+from datetime import datetime
+from logging import getLogger
+from pathlib import Path
+from typing import Any, Tuple, AnyStr, Dict
+from urllib.request import getproxies
+from .compat import IS_WINDOWS, urlparse
+from .cursor import SnowflakeCursor
+from .vendored import urllib3
+
+logger = getLogger(__name__)
+
+if IS_WINDOWS:
+    import winreg
+
+
+class ConnectionDiagnostic:
+    """Implementation of a connection test utility for Snowflake connector
+
+    Use new ConnectionTest() to get the object.
+    """
+
+    def __init__(self,
+        account: str,
+        host: str,
+        connection_diag_log_path: str,
+        connection_diag_whitelist_path: str,
+        proxy_host: str,
+        proxy_port: str,
+        proxy_user: str,
+        proxy_password: str,
+    ):
+        self.account = account
+        self.host = host
+        self.test_results: Dict[str, Any] = {'INITIAL': [], 'PROXY': [], 'SNOWFLAKE_URL': [], 'STAGE': [],
+                                             'OCSP_RESPONDER': [], 'OUT_OF_BAND_TELEMETRY': [], 'IGNORE': []}
+        host_type: str = "INITIAL"
```

Comment:
When you assign a constant you can skip the type-hint unless you want to widen the automatic type-hint (for example if this was possible set to `None` later).
```suggestion
        host_type = "INITIAL"
```
But it's not critical that you change this.